tests/testthat/test-list-combine.R

```diff
@@ -70,6 +70,15 @@ test_that("NULLs are ignored", {
   expect_equal(list_cbind(list(df1, NULL, df2)), vec_cbind(df1, df2))
 })
 
+test_that("NULLs are converted to NA when keep_empty = TRUE", {
+  df1 <- data.frame(x = 1)
+  df2 <- data.frame(y = 1)
+
+  expect_equal(list_c(list(1, NULL, 2), keep_empty = TRUE), c(1, NA, 2))
+  expect_equal(list_rbind(list(df1, NULL, df1), keep_empty = TRUE), vec_rbind(df1, NA, df1))
```

Comment:
I think I'd find it a little easier to verify that this is correct if the expected value looked like `data.frame(x = c(1, NA, 1)`